pkg/client/cache/reflector.go

```diff
@@ -337,16 +304,23 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			}
 			return nil
 		}
-		if err := r.watchHandler(w, &resourceVersion, resyncCh, stopCh); err != nil {
-			if err != errorResyncRequested && err != errorStopRequested {
+
+		if err := r.watchHandler(w, &resourceVersion, stopCh); err != nil {
+			if err != errorStopRequested {
 				glog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedType, err)
 			}
 			return nil
 		}
-		if r.canForceResyncNow() {
-			glog.V(4).Infof("%s: next resync planned for %#v, forcing now", r.name, r.nextResync)
-			return nil
-		}
+		/*
```

Comment:
@rrati 

> Maybe my understanding is wrong. I didn't think etcd cached anything, so if an object is on resource version 10 and I start a watch on resource version 9 then I will not get a notification for the change from version 9->10. I will get notice on version 11, but the change from 9->10 is "lost" to me.

This is wrong. If you start watch on 9 again, etcd will send change of 10 if the cache window for 10 is still there or etcd will return window clear error.